### PR TITLE
Use fixture

### PR DESCRIPTION
**Pull request by wli3**
Per code review to use fixture to reuse asset on disk

### CODE REVIEW

File: src/Tests/Microsoft.NET.ToolPack.Tests/DefaultPackAndResultNugetPackageNuGetPackageFixture.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                    nameof(DefaultPackWithShimsAndResultNugetPackageNuGetPackagexFixture),
+                    targetFramework);
+                assetMap[mapKey] = package;
+                return package;
+            }
+        }
+
+        private string SetupNuGetPackage(
+            bool multiTarget,
+            string uniqueName,
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.ToolPack.Tests/DefaultPackAndResultNugetPackageNuGetPackageFixture.cs]: I'd remove or rename the `uniqueName` parameter here.  It looks like it's not unique, it's always `nameof(DefaultPackWithShimsAndResultNugetPackageNuGetPackagexFixture)`, and it's different than the `uniqueName` praameter to the CreateTestAsset method.

File: src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToDefaultPackAToolProjectWithPackagedShim.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+using NuGet.Packaging;
+using NuGet.Frameworks;
+using fixture = Microsoft.NET.ToolPack.Tests.DefaultPackWithShimsAndResultNugetPackageNuGetPackagexFixture;
+
+namespace Microsoft.NET.ToolPack.Tests
+{
+    public class GivenThatWeWantToDefaultPackAToolProjectWithPackagedShim : SdkTest, IClassFixture<fixture>
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToDefaultPackAToolProjectWithPackagedShim.cs]: Nit: I would use a PascalCased alias for the fixture (and probably make it slightly more descriptive)

File: src/Tests/Microsoft.NET.ToolPack.Tests/DefaultPackAndResultNugetPackageNuGetPackageFixture.cs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        public TestAssetsManager TestAssetsManager { get; private set; }
+        public void Init(ITestOutputHelper log, TestAssetsManager testAssetsManager)
+        {
+            Log = log;
+            TestAssetsManager = testAssetsManager;
+        }
+        public Dictionary<(bool multiTarget, string targetFramework), string> assetMap
+            = new Dictionary<(bool multiTarget, string targetFramework), string>();
+
+        public string GetAsset(bool multiTarget, string targetFramework)
```
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.ToolPack.Tests/DefaultPackAndResultNugetPackageNuGetPackageFixture.cs]: I'd rename this method to make it clear that it is returning the path to a .nupkg, perhaps `GetTestToolPackagePath`.

File: src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToDefaultPackAToolProjectWithPackagedShim.cs (hunk 2 above)
Comment (dsplaisted) [reply src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToDefaultPackAToolProjectWithPackagedShim.cs]: I'm not sure what the difference between the new class name (`GivenThatWeWantToDefaultPackAToolProjectWithPackagedShim`) and the existing one (`GivenThatWeWantToPackAToolProjectWithPackagedShim`).  What does it mean to "default pack" a tool project?

I would suggest just leaving all these tests in the same class.  Some of the tests won't use the fixture, but that's fine.

**Review by dsplaisted — APPROVED** [review]

File: src/Tests/Microsoft.NET.ToolPack.Tests/DefaultPackAndResultNugetPackageNuGetPackageFixture.cs (hunk 1 above)
Comment (wli3) [reply src/Tests/Microsoft.NET.ToolPack.Tests/DefaultPackAndResultNugetPackageNuGetPackageFixture.cs]: I removed that param. It does not need to pass in

File: src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToDefaultPackAToolProjectWithPackagedShim.cs (hunk 2 above)
Comment (wli3) [reply src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToDefaultPackAToolProjectWithPackagedShim.cs]: Change the fixture name to NupkgOfPackWithShimsFixture and not use alias. And I merged it into GivenThatWeWantToPackAToolProjectWithPackagedShim

File: src/Tests/Microsoft.NET.ToolPack.Tests/DefaultPackAndResultNugetPackageNuGetPackageFixture.cs (hunk 3 above)
Comment (wli3) [reply src/Tests/Microsoft.NET.ToolPack.Tests/DefaultPackAndResultNugetPackageNuGetPackageFixture.cs]: Renamed to suggested

**Pull request merged**